diff -ruN Midnight.sol Midnight.sol
--- Midnight.sol	2026-07-01 17:13:29.648365974 +0200
+++ Midnight.sol	2026-07-01 17:15:21.172547427 +0200
@@ -407,8 +407,8 @@
         Position storage buyerPos = position[id][buyer];
         Position storage sellerPos = position[id][seller];
 
-        if (buyerPos.credit > 0 || units > buyerPos.debt) _updatePosition(offer.market, id, buyer);
-        if (sellerPos.credit > 0) _updatePosition(offer.market, id, seller);
+        if (hasCredit(id, buyer) || units > buyerPos.debt) _updatePosition(offer.market, id, buyer);
+        if (hasCredit(id, seller)) _updatePosition(offer.market, id, seller);
 
         uint256 buyerCreditIncrease = UtilsLib.zeroFloorSub(units, buyerPos.debt);
         uint256 sellerCreditDecrease = UtilsLib.min(units, sellerPos.credit);
@@ -455,28 +455,6 @@
         address payer = buyerCallback != address(0) ? buyerCallback : (offer.buy ? buyer : msg.sender);
         address receiver = offer.buy ? receiverIfTakerIsSeller : offer.receiverIfMakerIsSeller;
 
-        emit EventsLib.Take(
-            msg.sender,
-            keccak256(abi.encode(offer)),
-            id,
-            offer.buy,
-            offer.maker,
-            offer.group,
-            offer.ratifier,
-            ratifierData,
-            units,
-            taker,
-            buyerAssets,
-            sellerAssets,
-            newConsumed,
-            buyerPendingFeeIncrease,
-            sellerPendingFeeDecrease,
-            buyerCreditIncrease,
-            sellerCreditDecrease,
-            receiver,
-            payer
-        );
-
         bool wasLocked = UtilsLib.tExchange(LIQUIDATION_LOCK_SLOT, id, seller, true);
         if (buyerCallback != address(0)) {
             bytes memory buyerCallbackData = offer.buy ? offer.callbackData : takerCallbackData;
@@ -930,6 +908,10 @@
         return (feeLower * (end - timeToMaturity) + feeUpper * (timeToMaturity - start)) / (end - start);
     }
 
+    function hasCredit(bytes32 id, address user) internal view returns (bool) {
+        return position[id][user].credit > 0;
+    }
+
     /// @dev Reverts if the id is not a valid id of a touched market.
     /// @dev Returns the market corresponding to the given id.
     function toMarket(bytes32 id) external view returns (Market memory) {
